src/main/java/com/google/cloud/spanner/pgadapter/utils/MutationWriter.java

```diff
@@ -186,39 +195,32 @@ public int writeToSpanner(ConnectionHandler connectionHandler) throws SQLExcepti
     // Reset mutations, mutation counter, and batch size count for a new batch
     this.mutations = new ArrayList<>();
     this.mutationCount = 0;
-    this.batchSize = 0;
     return this.rowCount;
   }
 
-  public void rollback(ConnectionHandler connectionHandler, byte[] payload) throws Exception {
+  public void rollback(ConnectionHandler connectionHandler) throws Exception {
     Connection connection = connectionHandler.getJdbcConnection();
     connection.rollback();
     this.mutations = new ArrayList<>();
     this.mutationCount = 0;
-    this.batchSize = 0;
-    createErrorFile(payload);
+    writeToErrorFile(this.payload.toByteArray());
   }
 
-  public void createErrorFile(byte[] payload) throws IOException {
+  private void createErrorFile() throws IOException {
     File unsuccessfulCopy = new File(ERROR_FILE);
-    if (unsuccessfulCopy.createNewFile()) {
-      this.fileWriter = new FileWriter(ERROR_FILE);
-      writeToErrorFile(payload);
-    } else {
-      System.err.println("File " + unsuccessfulCopy.getName() + " already exists");
-    }
+    this.fileWriter = new FileWriter(unsuccessfulCopy, false);
   }
 
   public void writeToErrorFile(byte[] payload) throws IOException {
```

Comment:
rename to: writeCopyDataToErrorFile

add a comment that this is used if there are problems while collecting CopyData messages or building the mutation list

Reply:
Done.